eitprocessing/eit_data/sentec.py

```diff
@@ -18,4 +34,144 @@ def _from_path(  # pylint: disable=too-many-arguments
         max_frames: int | None = None,
         return_non_eit_data: bool = False,
     ) -> Self | tuple[Self, ContinuousDataCollection, SparseDataCollection]:
-        ...
+
+        with open(path, "br") as fh:
+            # Find the length of the file
+            fh.seek(0, 2)
+            file_length = fh.tell()
+
+            # go back to the beginning of the file
+            fh.seek(0, 0)
+
+            # instantiate reader
+            reader = Reader(fh)
+
+            # read the version int8
+            version = reader.unsigned_char('little')
+
+            if version < 2:
+                warnings.warn(f'File version {version}. Version 2 or higher expected.')
+
+            time = []
+            image = None
+            index = 0
+
+            # while there are still data to be read and the number of read data points is higher
+            # than the maximum specified, keep reading
+            while fh.tell() < file_length and (max_frames is None or len(time) < max_frames):
+
+                # Read time stamp uint64
+                timestamp = reader.unsigned_long_long('little')
+                # Read DomainId uint8
+                domain_id = reader.unsigned_char('little')
+                # read number of data fields uint8
+                number_data_fields = reader.unsigned_char('little')
+
+                for _ in range(number_data_fields):
+                    # read data id uint8
+                    data_id = reader.unsigned_char('little')
+                    # read payload size ushort
+                    payload_size = reader.unsigned_short('little')
+
+                    if payload_size != 0:
+                        # read frame (domain 16 = measurements, data 5 = zero_ref_image)
+                        if domain_id == 16 and data_id == 5:
+                            index += 1
+
+                            ref = cls._read_frame(fh, index, payload_size, reader, first_frame)
+
+                            if ref is not None:
+                                image = np.concatenate([image, ref[np.newaxis, :, :]], axis=0) \
+                                    if image is not None else ref[np.newaxis, :, :]
+
+                                time.append(timestamp)
+
+                        # read the framerate from the file, if present
+                        # (domain 64 = configuration, data 5 = framerate)
+                        elif domain_id == 64 and data_id == 1:
+                            framerate = reader.float32('little')
```

Comment:
Is this always present? If so, the function should not accept a framerate as
argument, because it will always be overridden.

Reply:
I expect this to be present, but I don't know for sure. I left this in the argument as well, as I don't know for sure is by default stored in the file. But, if this information is available in the data, I would use that.

Reply:
I think we should warn the user if the given framerate is overridden by this data point.